language/documentation/book/src/unit-testing.md

```diff
@@ -0,0 +1,332 @@
+
+#[test_only] // .. and functions. Can only be called from test code, but not a test
+fun test_only_function(...) { ... }
+```
+
+## Running Unit Tests
+
+Unit tests for a Move package can be run with the [`move package test`
+command](./packages.md).
+
+When running tests, every test will either `PASS`, `FAIL`, or `TIMEOUT`. If a test case fails, the location of the failure along with the function name that caused the failure will be reported if possible. You can see an example of this below.
+
+A test will be marked as timing out if it exceeds the maximum number of instructions that can be executed for any single test. This bound can be changed using the options below, and its default value is set to 5000 instructions. Additionally, while the result of a test is always deterministic, tests are run in parallel by default, so the ordering of test results in a test run is non-deterministic unless running with only one thread (see `OPTIONS` below).
+
+There are also a number of options that can be passed to the unit testing binary to fine-tune testing, and to help debug failing tests. These are:
+
+```
+SAGE:
+    move package test [FLAGS] [OPTIONS]
+
+FLAGS:
+        --stackless         Use the stackless bytecode interpreter to run the tests and cross check its results with the
+                            execution result from Move VM
+        --coverage
+    -g, --state_on_error    Show the storage state at the end of execution of a failing test
+    -h, --help              Prints help information
+    -l, --list              List all tests
+    -s, --statistics        Report test statistics at the end of testing
+    -V, --version           Prints version information
+    -v                      Print additional diagnostics
+        --verbose           Verbose mode
+
+OPTIONS:
+    -f, --filter <filter>                A filter string to determine which unit tests to run
+    -i, --instructions <instructions>    Bound the number of instructions that can be executed by any one test [default:
+                                         5000]
+    -t, --threads <num_threads>          Number of threads to use for running tests [default: 8]
+    -p, --path <package-path>            Path to a package which the command should be run with respect to. Defaults to
+                                         current directory [default: .]
+        --storage-dir <storage-dir>      Directory storing Move resources, events, and module bytecodes produced by
+                                         module publishing and script execution [default: storage]
+```
+
+While each of these flags and options are fairly self-explanatory, it is worth mentioning that when filtering tests with the `-f` option, any test whose fully qualified name (i.e., `address::module::function_name`) contains the `<filter>` string will be run. Additionally, the `--coverage` flag collects test coverage information that you can view using the `move package coverage` command.
```

Comment:
There are a few documents in our repository that include the output from tools' `--help` options, and I've found many of them to be out of date (it's also hard to keep them in sync -- do I send pull requests to update them? That means I need to get reviews, wait for bors to land the code... and anyone who changes the options or their help text needs to remember to update any docs that includes help text). Instead, how about the following?

```suggestion
There are also a number of options that can be passed to the unit testing binary to fine-tune testing, and to help debug failing tests. You may view these by running `move package test --help`.
```

In this case, though, I would suggest that the additional notes you include here ought to be included in the documentation of the options. That is, the `-f` option help text should include the fact that any test that includes the filter string will be run, and the `--coverage` option help text should indicate that its output can be viewed with the `move package coverage` command -- if they don't already, that is.

Reply:
Very good point! I'll update this, and add the additional documentation to the commands

Reply:
I've removed this, and updated the documentation comments around the package commands in the bottom commit of this PR.